packages/js/src/matcher.ts

```diff
@@ -25,19 +25,63 @@ function matchLogic(logic: any, text: string): boolean {
   return false
 }
 
+/**
+ * Find a provider by matching against provider_match logic
+ * @param providers - List of available providers
+ * @param providerId - The provider ID to match
+ * @returns The matching provider or undefined
+ */
+export function findProviderByMatch(providers: Provider[], providerId: string): Provider | undefined {
+  const normalizedProviderId = providerId.toLowerCase().trim()
+
+  // First try exact match by ID
+  const exactMatch = providers.find((p) => p.id === normalizedProviderId)
+  if (exactMatch) return exactMatch
+
+  // Then try provider_match logic
+  return providers.find((p) => p.provider_match && matchLogic(p.provider_match, normalizedProviderId))
+}
+
+/**
+ * Normalize a model name based on provider and model patterns
+ * @param providerId - The normalized provider ID
+ * @param modelName - The raw model name
+ * @returns The normalized model name
+ */
+export function normalizeModel(providerId: string, modelName: string): string {
```

Comment:
I really don't think we need this.

Reply:
yeah, it is only for internal usage, I'll remove `export`